src/streaming/Stream.js

```diff
@@ -52,66 +52,124 @@ MediaPlayer.dependencies.Stream = function () {
-                            msg = type + "Codec (" + codec + ") is not supported.";
-                            self.errHandler.manifestError(msg, "codec", manifest);
-                            self.log(msg);
-                            return;
-                        }
+            return true;
+        },
+
+        onCurrentTrackChanged = function(e) {
+            var processor = getProcessorForMediaInfo.call(this, e.data.oldMediaInfo);
+            if (!processor) return;
+
+            var currentTime = this.playbackController.getTime(),
+                buffer = processor.getBuffer(),
+                mediaInfo = e.data.newMediaInfo,
+                manifest = this.manifestModel.getValue(),
+                idx = streamProcessors.indexOf(processor),
+                mediaSource = processor.getMediaSource();
+
+            if (mediaInfo.type !== "fragmentedText"){
+                processor.reset(true);
+                createStreamProcessor.call(this, mediaInfo, manifest, mediaSource, {buffer: buffer, replaceIdx: idx, currentTime: currentTime});
+                this.playbackController.seek(this.playbackController.getTime());
+            }else {
+                processor.setIndexHandlerTime(currentTime);
+                processor.updateMediaInfo(manifest, mediaInfo);
+            }
+        },
+
+        createStreamProcessor = function(mediaInfo, manifest, mediaSource, optionalSettings) {
+            var self = this,
+                streamProcessor = self.system.getObject("streamProcessor"),
+                allMediaForType = this.adapter.getAllMediaInfoForType(manifest, streamInfo, mediaInfo.type);
+
+            streamProcessor.initialize(getMimeTypeOrType.call(self, mediaInfo), self.fragmentController, mediaSource, self, eventController);
+            self.abrController.updateTopQualityIndex(mediaInfo);
+
+            if (optionalSettings) {
+                streamProcessor.setBuffer(optionalSettings.buffer);
+                streamProcessors[optionalSettings.replaceIdx] = streamProcessor;
+                streamProcessor.setIndexHandlerTime(optionalSettings.currentTime);
+            }
+
+            if((mediaInfo.type === "text" || mediaInfo.type === "fragmentedText")) {
+                var idx;
+                for(var i = 0; i < allMediaForType.length; i++){
+                    if(allMediaForType[i].index === mediaInfo.index) {
+                        idx = i;
                     }
+                    streamProcessor.updateMediaInfo(manifest, allMediaForType[i]);//creates text tracks for all adaptations in one stream processor
+                }
+                if(mediaInfo.type === "fragmentedText"){
+                    streamProcessor.updateMediaInfo(manifest, allMediaForType[idx]);//sets the initial media info
                 }
+            }else {
+                streamProcessor.updateMediaInfo(manifest, allMediaForType[0]);
```

Comment:
@AkamaiDASH, I believe this line breaks the adaptation switch process since it forces the player to always pick the first adaptation instead of the one passed as an argument. I created a PR with a fix, but before merging it I want to get your confirmation that it does not confront with your intentions that I am not aware of:
#700
After review, if you think that everything is fine, could you please merge the PR since the fixes that it includes are quite important.